langchain/agents/agent.py

```diff
@@ -718,6 +734,10 @@ def _take_next_step(
                     color=color,
                     **tool_run_kwargs,
                 )
+            elif agent_action.tool == "Exception":
+                # just raise exception then, since tool
+                # wasn't installed
+                raise ValueError(agent_action.tool_input)
```

Comment:
Would be helpful to pass the stack-trace somehow for this flow since `str(e)` swallows helpful information